llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -6725,9 +6725,9 @@ void BoUpSLP::buildTree_rec(ArrayRef<Value *> VL, unsigned Depth,
   bool IsScatterVectorizeUserTE =
       UserTreeIdx.UserTE &&
       UserTreeIdx.UserTE->State == TreeEntry::ScatterVectorize;
-  bool AreAllSameInsts =
-      (S.getOpcode() && allSameBlock(VL)) ||
-      (S.OpValue->getType()->isPointerTy() && IsScatterVectorizeUserTE &&
+  bool AreAllSameBlock = (S.getOpcode() && allSameBlock(VL));
```

Comment:
Remove extra parens

Reply:
Thanks, fixed.

Reply:
Since I don't have write permissions, could you help me merge?